File: dungeons-26.1.2/src/main/java/fishmod/utils/config/FishConfig.java
package fishmod.utils.config;

import fishmod.features.dungeon.FishEstTotal;
import fishmod.features.dungeon.FishPuzzleDisplay;
import fishmod.utils.config.values.FishSettings;
import config.practical.manager.ConfigManager;

import java.util.List;

/**
 * Separate config manager for FishMod-specific settings.
 * Stored in config/fishmod-settings.json, independent of blade-addons config.
 */
public class FishConfig {

    public static final ConfigManager manager = new ConfigManager(
            "config/fishmod-settings.json",
            List.of(FishSettings.class, FishPuzzleDisplay.class, FishEstTotal.class,
                    fishmod.utils.dungeon.Phase.class,
                    fishmod.utils.dungeon.Section.class,
                    fishmod.utils.dungeon.Split.class,
                    fishmod.utils.config.values.Dungeons.class,
                    fishmod.utils.config.values.Floor7.class,
                    fishmod.utils.config.values.Buttons.class,
                    fishmod.features.dungeon.f7.F7Huds.class,
                    fishmod.utils.config.values.DungeonMapSettings.class,
                    fishmod.features.dungeon.map.DungeonMapHud.class));
}

File: dungeons-26.1.2/src/main/kotlin/fishmod/utils/config/FishConfig.kt
package fishmod.utils.config

import config.practical.manager.ConfigManager
import fishmod.features.dungeon.FishEstTotal
import fishmod.features.dungeon.FishPuzzleDisplay
import fishmod.features.dungeon.f7.F7Huds
import fishmod.utils.config.values.Buttons
import fishmod.utils.config.values.Dungeons
import fishmod.utils.config.values.DungeonMapSettings
import fishmod.utils.config.values.FishSettings
import fishmod.utils.config.values.Floor7
import fishmod.utils.dungeon.Phase
import fishmod.utils.dungeon.Section
import fishmod.utils.dungeon.Split

/**
 * Separate config manager for FishMod-specific settings.
 * Stored in config/fishmod-settings.json, independent of blade-addons config.
 */
object FishConfig {

    @JvmField
    val manager: ConfigManager = ConfigManager(
        "config/fishmod-settings.json",
        listOf(
            FishSettings::class.java, FishPuzzleDisplay::class.java, FishEstTotal::class.java,
            Phase::class.java,
            Section::class.java,
            Split::class.java,
            Dungeons::class.java,
            Floor7::class.java,
            Buttons::class.java,
            F7Huds::class.java,
            DungeonMapSettings::class.java
        )
    )
}
